server/command/service.go

```diff
@@ -261,60 +261,65 @@ func (s *service) ExecuteCommand(pluginContext *plugin.Context, commandArgs *mod
-			Other: "invalid arguments to command.Handler. Please contact your system administrator",
-		})))
+		return s.errorOut(params, utils.NewLocError(&nsi18n.LocalizeConfig{
+			DefaultMessage: &nsi18n.Message{
+				ID:    "apps.command.error.invalidArguments",
+				Other: "invalid arguments to command.Handler. Please contact your system administrator",
+			},
+		}), errors.New("invalid arguments to command.Handler. Please contact your system administrator"))
 	}
 
 	conf := s.conf.MattermostConfig().Config()
 	enableOAuthServiceProvider := conf.ServiceSettings.EnableOAuthServiceProvider
 	if enableOAuthServiceProvider == nil || !*enableOAuthServiceProvider {
-		return s.errorOut(params, errors.New(s.conf.I18N().LocalizeWithConfig(loc, &nsi18n.LocalizeConfig{
+		url := commandArgs.SiteURL + "/admin_console/integrations/integration_management"
+		return s.errorOut(params, utils.NewLocError(&nsi18n.LocalizeConfig{
 			DefaultMessage: &nsi18n.Message{
 				ID:    "apps.command.enableOAuth",
 				Other: "the system setting `Enable OAuth 2.0 Service Provider` needs to be enabled in order for the Apps plugin to work. Please go to {{.URL}} and enable it.",
 			},
 			TemplateData: map[string]string{
-				"URL": commandArgs.SiteURL + "/admin_console/integrations/integration_management",
+				"URL": url,
 			},
-		})))
+		}), fmt.Errorf("the system setting `Enable OAuth 2.0 Service Provider` needs to be enabled in order for the Apps plugin to work. Please go to %s and enable it.", url))
 	}
 
 	enableBotAccounts := conf.ServiceSettings.EnableBotAccountCreation
 	if enableBotAccounts == nil || !*enableBotAccounts {
-		return s.errorOut(params, errors.New(s.conf.I18N().LocalizeWithConfig(loc, &nsi18n.LocalizeConfig{
+		url := commandArgs.SiteURL + "/admin_console/integrations/bot_accounts"
+		return s.errorOut(params, utils.NewLocError(&nsi18n.LocalizeConfig{
 			DefaultMessage: &nsi18n.Message{
 				ID:    "apps.command.enableBot",
 				Other: "the system setting `Enable Bot Account Creation` needs to be enabled in order for the Apps plugin to work. Please go to {{.URL}} and enable it.",
 			},
 			TemplateData: map[string]string{
-				"URL": commandArgs.SiteURL + "/admin_console/integrations/bot_accounts",
+				"URL": url,
 			},
-		})))
+		}), fmt.Errorf("the system setting `Enable Bot Account Creation` needs to be enabled in order for the Apps plugin to work. Please go to %s and enable it.", url))
 	}
 
 	split := strings.Fields(commandArgs.Command)
 	if len(split) < 2 {
-		return s.errorOut(params, errors.New(s.conf.I18N().LocalizeDefaultMessage(loc, &nsi18n.Message{
-			ID:    "apps.command.error.noSubCommand",
-			Other: "no subcommand specified, nothing to do",
-		})))
+		return s.errorOut(params, utils.NewLocError(&nsi18n.LocalizeConfig{
+			DefaultMessage: &nsi18n.Message{
+				ID:    "apps.command.error.noSubCommand",
+				Other: "no subcommand specified, nothing to do",
+			},
+		}), errors.New("no subcommand specified, nothing to do"))
```

Comment:
Should this even return a `error`  given that the output is specifically targeting the user? I.e. should errors that "overlap" be reported as two different one or should the generic be an extension of the localized?

Reply:
We have the option in the ErrorOut for the command. And the error is not used either way. The only reason I saw to keep it here was more semantic. If we were to pass nil in this error, we may consider there is no error. If we consider an error happening every time either locError or error is != nil, we would need to update all the checks (which may be none in this particular case).

Reply:
I couldn't find a place where the locError wrapping is used so I don't think any checks need to be changed right now. `ErrorOut` already checks for both errors when outputting them. 1/5 that not overlapping the errors would also make it simpler for developers to think about them and the code would be cleaner as the information is only in one place.

For example 

```go
	m, err := apps.ManifestFromJSON(data)
	if err != nil {
		return errorOut(params, errors.Wrap(err, "unable to decode "+manifestURL))
		return s.errorOut(params, utils.NewLocError(&i18n.LocalizeConfig{
			DefaultMessage: &i18n.Message{
				ID:    "apps.command.install.http.error.decodeManifest",
				Other: "unable to decode {{.ManifestURL}}",
			},
			TemplateData: map[string]interface{}{
				"ManifestURL": manifestURL,
			},
		}), errors.Wrap(err, "unable to decode "+manifestURL))
	}
```

would be

```go
	m, err := apps.ManifestFromJSON(data)
	if err != nil {
		return errorOut(params, errors.Wrap(err, "unable to decode "+manifestURL))
		return s.errorOut(params, utils.NewLocError(&i18n.LocalizeConfig{
			DefaultMessage: &i18n.Message{
				ID:    "apps.command.install.http.error.decodeManifest",
				Other: "unable to decode {{.ManifestURL}}",
			},
			TemplateData: map[string]interface{}{
				"ManifestURL": manifestURL,
			},
		}), err
	}
```

as the locError contains all the error that was generated here.

Reply:
I agree with you 90%. The only caveat that comes to mind is one of the concerns you were having about the localization of errors, and that is the "searchability" of those. I will remove all the redundant information, but please take another look in case you want to put back anything into the "not localized string".